client/me/account/toggle-sites-as-landing-page.tsx

```diff
@@ -0,0 +1,53 @@
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch, useSelector } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { successNotice } from 'calypso/state/notices/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference, isSavingPreference } from 'calypso/state/preferences/selectors';
+
+type ToggleSitesLandingPageProps = {
+	onToggle?: ( checked: boolean ) => void;
+	className?: string;
+};
+
+function ToggleSitesAsLandingPage( { onToggle, className }: ToggleSitesLandingPageProps ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const useSitesAsLandingPage = useSelector(
+		( state ) => getPreference( state, 'sites-landing-page' )?.useSitesAsLandingPage
+	);
+	const isSaving = useSelector( ( state ) => isSavingPreference( state ) );
+
+	async function handleToggle( isChecked: boolean ) {
+		const preference = { useSitesAsLandingPage: isChecked, updatedAt: Date.now() };
+		dispatch(
+			recordTracksEvent( 'calypso_sites_as_landing_page_toggled', {
+				sites_as_landing_page: isChecked,
+			} )
+		);
+
+		await dispatch( savePreference( 'sites-landing-page', preference ) );
+
+		dispatch(
+			successNotice( translate( 'Settings saved successfully!' ), {
+				id: 'sites-landing-page-save',
+				duration: 10000,
+			} )
+		);
+		onToggle?.( isChecked );
+	}
+
+	return (
+		<div className={ className }>
+			<ToggleControl
+				checked={ !! useSitesAsLandingPage }
+				onChange={ handleToggle }
+				disabled={ isSaving }
```

Comment:
The build is failing here because the `@wordpress/components` type definitions don't include the `disabled` props 😭 This is always frustrating.

Here's a hack that doesn't require `@wordpress/components` to update before we can merge:
```typescript
import { ToggleControl as OriginalToggleControl } from '@wordpress/components';

const ToggleControl = OriginalToggleControl as React.ComponentType<
	OriginalToggleControl.Props & {
		disabled?: boolean;
	}
>;
```